File: apps/api/app/schema/job.py
from typing import List, Optional

from pydantic import BaseModel


class Job(BaseModel):
    company: str
    title: str
    description: str
    salary: Optional[str] = None
    responsibilities: Optional[List[str]] = None
    qualifications: Optional[List[str]] = None
    logistics: Optional[List[str]] = None
    location: Optional[List[str]] = None
    additional_info: Optional[List[str]] = None
    link: Optional[str] = None

File: src/lib/types/job.ts
export interface Job {
	company: string;
	title: string;
	description: string;
	salary?: string | null;
	responsibilities?: string[] | null;
	qualifications?: string[] | null;
	logistics?: string[] | null;
	location?: string[] | null;
	additional_info?: string[] | null;
	link?: string | null;
}

export interface JobDocument {
	type: 'resume' | 'cover' | 'research' | 'prep';
	content: string;
}
